#llm_service.proto
syntax = "proto3";
import "google/protobuf/struct.proto";

service LLMService {
  // Handshake to send and receive configuration
  rpc Handshake(HandshakeRequest) returns (HandshakeResponse);

  // Method for single text generation
  rpc GenerateText(TextRequest) returns (TextResponse);

  // Method for streaming text generation
  rpc StreamText(TextRequest) returns (stream TextResponse);

  // Method to delete an agent
  rpc DeleteAgent(DeleteAgentRequest) returns (DeleteAgentResponse);

}

message HandshakeRequest {
  string json_config = 1; // JSON configuration sent by the client
}

message HandshakeResponse {
  google.protobuf.Struct json_config = 1; // JSON configuration acknowledged by the server
}

message TextRequest {
  string input = 1;       // Input text
  string agent_name = 2;   // Agent name
  string user_id = 3; // UserID
}



message TextResponse {
  google.protobuf.Struct json_config = 1; // JSON configuration acknowledged by the server
}

message DeleteAgentRequest {
  string agent_name = 1; // The name of the agent to delete
}

message DeleteAgentResponse {
  google.protobuf.Struct json_config = 1; // Indicates if the deletion was successful
}